app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -334,6 +335,11 @@ public void requestPermissions(int actionId) {
                         throw new IllegalArgumentException("Unknown Action");
                 }
             }
+
+            @Override
+            public void permissionDeniedToast() {
```

Comment:
We have different permissions, location and storage. Pass actionId to this interface, see Snackbar makeAskAgainSnackBar(int actionId);

Reply:
fixed